packages/input/src/Input/Input.tsx

```diff
@@ -0,0 +1,157 @@
+
+const Root = React.forwardRef<RootRef, RootProps>(
+  (
+    {
+      size = "2",
+      variant = "outline",
+      className,
+      disabled,
+      error,
+      children,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement>(null);
+    const composedRefs = useComposedRefs(forwardedRef, inputRef);
+
+    const state = disabled ? "disabled" : error ? "error" : "default";
+
+    return (
+      <InputContext.Provider value={{ size, variant, state }}>
+        <div
+          className={clsx(
+            "box-border flex items-center transition-all",
+            "border-2 border-solid text-gray-12 placeholder:text-gray-10",
+            COLOR.Container[state][variant],
+            SIZE.Container[size],
+          )}
+          // Focus the input when clicking on the container
+          onPointerDown={(event) => {
+            const target = event.target as HTMLElement;
+
+            // Make sure we're not clicking on an interactive element
+            if (target.closest("button, a")) {
+              event.preventDefault();
+              return;
+            }
+
+            const input = inputRef.current;
+            if (!input) return;
+
+            requestAnimationFrame(() => {
+              input.focus();
+            });
+          }}
+        >
+          <input
+            className={clsx(
+              "appearance-none text-gray-12 border-none shadow-0 outline-0 bg-transparent",
+              "[font-family:inherit] h-full w-full",
+              "order-2",
+              SIZE.Input[size],
+              className,
+            )}
+            disabled={disabled}
+            {...props}
+            ref={composedRefs}
+          />
+          {children}
+        </div>
+      </InputContext.Provider>
+    );
+  },
+);
+
+type SlotProps = React.ComponentPropsWithoutRef<typeof RadixSlot> & {
+  size?: keyof typeof SIZE.Slot;
+  position?: "leading" | "trailing";
+};
+type SlotRef = React.ElementRef<typeof RadixSlot>;
+
+const Slot = React.forwardRef<SlotRef, SlotProps>(
+  ({ position = "leading", ...props }, forwardedRef) => {
+    const context = React.useContext(InputContext);
+    return (
+      <div
+        className={clsx(
+          "group box-border flex items-center justify-center h-full",
+          "[&>[data-tgph-button]]:w-full [&>[data-tgph-button]]:h-auto",
+          // Overrides to match the icon button in figma
+          "[&>[data-tgph-button-layout='icon-only']]:aspect-square [&>[data-tgph-button-layout='icon-only']]:p-0",
+          // Overrides default button layout to match the button in figma
+          "[&>[data-tgph-button-layout='default']]:px-2",
+          // If only an icon button is present, set the aspect ratio to square
+          "[&:has([data-tgph-button-layout='icon-only'])]:aspect-square",
+          // If only an icon button is present, reset the padding to spacing-1
+          "[&:has([data-tgph-button-layout='icon-only'])]:!p-1",
```

Comment:
Overrides some styles for the Button `icon-only` and `default` layouts so it fits into each size of the text field like figma outlines.